lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -183,7 +183,8 @@ def safe_add_concurrent_partitioned_index(
     using: nil,
     unique: nil,
     where: nil,
-    comment: nil
+    comment: nil,
+    recurse: nil
```

Comment:
When would we want `recurse` to be false?

Reply:
So I'm imagining a scenario where the user might want to break up the operation into several discrete method calls (or even multiple migrations). If you have sub-partitioning that is multiple levels deep, maybe you'd want to do:

```ruby
safe_add_concurrent_partitioned_index :foo_sub_level_2, :column, name_suffix: :column
safe_add_concurrent_partitioned_index :foo_sub_level_1, :column, name_suffix: :column, if_not_exists: true
safe_add_concurrent_partitioned_index :foo, :column, name_suffix: :column, if_not_exists: true
```

Of course, I think you could accomplish the same thing with:

```ruby
safe_add_concurrent_partitioned_index :foo_sub_level_2, :column, name_suffix: :column, recurse: true
safe_add_concurrent_partitioned_index :foo_sub_level_1, :column, name_suffix: :column, if_not_exists: true, recurse: true
safe_add_concurrent_partitioned_index :foo, :column, name_suffix: :column, if_not_exists: true, recurse: true
```

But there will be significantly more SQL queries being executed in the latter.

Maybe it makes sense to just default `recurse` to `true`, as I think that's what you'd want most of the time, and allow the user to opt out?

Reply:
Yeah, I like that better. I'm not 100% convinced we _need_ the knob, but I don't really know if I'd want it in the future or not, so I think it's quite reasonable to add it. But doing the reasonable default seems like a good thing up-front.